lib/transformations/removeDeprecatedPlugins/removeDeprecatedPlugins.js

```diff
@@ -1,21 +1,48 @@
-const findPluginsByName = require('../utils').findPluginsByName;
+const codeFrame = require('babel-code-frame');
+const chalk = require('chalk');
+const utils = require('../utils');
 
-module.exports = function(j, ast) {
+const example = `plugins: [
+    new webpack.optimize.OccurrenceOrderPlugin(),
+    new webpack.optimize.UglifyJsPlugin(),
+    new webpack.optimize.DedupePlugin()
+]`;
+
+module.exports = function(j, ast, source) {
 	// List of deprecated plugins to remove
 	// each item refers to webpack.optimize.[NAME] construct
 	const deprecatedPlugingsList = [
 		'webpack.optimize.OccurrenceOrderPlugin',
 		'webpack.optimize.DedupePlugin'
 	];
 
-	return findPluginsByName(j, ast, deprecatedPlugingsList)
+	return utils.findPluginsByName(j, ast, deprecatedPlugingsList)
 		.forEach(path => {
-			// Check how many plugins are defined and
-			// if there is only last plugin left remove `plugins: []` completely
-			if (path.parent.value.elements.length === 1) {
-				j(path.parent.parent).remove();
+			// For now we only support the case there plugins are defined in an Array
+			if (path.parent &&
+				path.parent.value &&
+				utils.isType(path.parent.value, 'ArrayExpression')
+			) {
+				// Check how many plugins are defined and
+				// if there is only last plugin left remove `plugins: []` node
+				if (path.parent.value.elements.length === 1) {
+					j(path.parent.parent).remove();
```

Comment:
Not sure what you mean by "valid way" but this works exactly as if I would come from the top of the tree.